builder/azure/arm/config_test.go

```diff
@@ -51,6 +64,19 @@ func TestConfigShouldProvideReasonableDefaultValues(t *testing.T) {
 	}
 }
 
+func TestConfigForCustomVHDSource(t *testing.T) {
```

Comment:
Maybe another test case to that mixes image_uri and image_publisher, image_offer, image_sku to match the additional config checks you make.